deep-learning/_posts/2018-06-07-attention.md

```diff
@@ -0,0 +1,43 @@
+---
+layout: review
+title:  "Show, Attend and Tell: Neural Image Caption Generation with Visual Attention"
+tags:   deep-learning CNN RNN LSTM essentials
+author: "Pierre-Marc Jodoin"
+pdf: "https://arxiv.org/pdf/1502.03044.pdf"
+cite:
+  authors: "Kelvin Xu, Jimmy Lei Ba, Ryan Kiros, Kyunghyun Cho, Aaron Courville, Ruslan Salakhutdinov, Richard S. Zemel,Yoshua Bengio "
+  title:   "Show, Attend and Tell: Neural Image Caption Generation with Visual Attention"
+  venue:   "ICML 2015"
+---
+
+## Introduction
+
+One of the most highly cited attention paper.  The goal is to generate a caption (sequence of words) from an image (a 2D array of pixels). As mentioned in [this blog](https://shadowthink.com/blog/research/2017/06/11/review-image-captioning#m-rnn), this method generalizes over previous recurrent image-to-text methods.
+
+# Description
+
+As shown in the following 2 figures
+
+
+![](/deep-learning/images/attention/sc01.png)
+
+![](/deep-learning/images/attention/sc04.png)
+
+
+
+ the method processes an image with a CNN.  This results into a set of $$14 \times 14 \times 512$$ feature maps after the fourth convolutional layer.  These feature maps are then fed to an LSTM.  The output of this LSTM is two fold : 1) the prediction of a word and 2) a $$14 \times 14$$ mask with values between 0 and 1.  This mask is called the **attention** associated to the predicted work.  The attention mask is then multiplied element wise to the feature maps which is fed back into the input of the LSTM.  The system stops iterating when the LSTM generates the symbol "EOS" (End of Sentence).
```

Comment:
associated to the predicted **word**

Reply:
Are you sure the mask correspond to the word that has just been predicted? Isn't it the prediction of where the object of the next word might be? If the current word is pizza, and the network masks everything except the pizza, and feed this new image as input for the next timestep, how can the network predict the next word?